tests/actions/ReportPreviewActionUtilsTest.ts

```diff
@@ -0,0 +1,155 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import getReportPreviewAction from '@libs/ReportPreviewActionUtils';
+// eslint-disable-next-line no-restricted-syntax
+import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Policy, Report, ReportViolations, Transaction, TransactionViolation} from '@src/types/onyx';
+
+const CURRENT_USER_ACCOUNT_ID = 1;
+const CURRENT_USER_EMAIL = 'tester@mail.com';
+
+const SESSION = {
+    email: CURRENT_USER_EMAIL,
+    accountID: CURRENT_USER_ACCOUNT_ID,
+};
+
+const PERSONAL_DETAILS = {
+    accountID: CURRENT_USER_ACCOUNT_ID,
+    login: CURRENT_USER_EMAIL,
+};
+
+const REPORT_ID = 1;
+const TRANSACTION_ID = 'TRANSACTION_ID';
+const REPORT_TRANSACTIONS = {} as Transaction[];
+const VIOLATIONS = {} as OnyxCollection<TransactionViolation[]>;
+
+jest.mock('@libs/ReportUtils', () => ({
+    ...jest.requireActual<typeof ReportUtils>('@libs/ReportUtils'),
+    hasViolations: jest.fn().mockReturnValue(false),
+}));
+describe('getReportPreviewAction', () => {
+    beforeAll(() => {
+        Onyx.init({
+            keys: ONYXKEYS,
+        });
+    });
+
+    beforeEach(async () => {
+        Onyx.clear();
+        await Onyx.merge(ONYXKEYS.SESSION, SESSION);
+        await Onyx.set(ONYXKEYS.PERSONAL_DETAILS_LIST, {[CURRENT_USER_ACCOUNT_ID]: PERSONAL_DETAILS});
+    });
+
+    it('canSubmit should return true for expense preview report with manual submit', async () => {
+        const report = {
+            reportID: REPORT_ID,
+            type: CONST.REPORT.TYPE.EXPENSE,
+            ownerAccountID: CURRENT_USER_ACCOUNT_ID,
+            stateNum: CONST.REPORT.STATE_NUM.OPEN,
+            statusNum: CONST.REPORT.STATUS_NUM.OPEN,
+        } as unknown as Report;
+        const policy = {
+            autoReportingFrequency: CONST.POLICY.AUTO_REPORTING_FREQUENCIES.IMMEDIATE,
+        };
```

Comment:
We should create the policy first, then modify the value and we can remove the data type cast  `policy as Policy`

```suggestion
        const policy = createRandomPolicy(0);
        policy.autoReportingFrequency = CONST.POLICY.AUTO_REPORTING_FREQUENCIES.IMMEDIATE;
```

Please apply the same change to other test cases.